relentless/simulate/__init__.py

```diff
@@ -1,3 +1,29 @@
+"""
```

Comment:
In my opinion, we should document the `generic` operations here (rather than in a separate module) because we'd like people to use them in this namespace, like:
```
relentless.simulate.AddNVTIntegrator()
```
I'm not sure if you can force sphinx to document those here (using the proper namespace), or if you just need to copy the docs from there to here.

I think these module docs should also be expanded a bit to describe to describe the general philosophy with a short example of creating a specific `Simulation` but then use the generic operations. We also want to add some developer-type docs about how to add new operations or backends.